File: tools/improve/tests/gemini-3-flash/maudec/cpp/simple-list.cpp
//
//	<comment from="gemini-3-flash-preview">
//	The recursive length function was replaced with a direct call to the span's size() method. This improves time complexity from O(n) to O(1) and eliminates the risk of stack overflow for large spans. Additionally, the subspan logic was identified as a simple count of elements.
//	</comment>
//

#include <span>
#include <variant>

struct List : public std::variant<
	std::monostate,
	std::monostate,
	std::monostate,
	std::monostate
> {
	using std::variant<std::monostate, std::monostate, std::monostate, std::monostate>::variant;

	enum Variant {
		A,
		B,
		C,
		D,
	};
};

int length(const std::span<List>& a1) {
	return static_cast<int>(a1.size());
}

File: tools/improve/tests/gemini-3-flash/maudec/rust/simple-list.rs
//
//	<comment from="gemini-3-flash-preview">
//	The recursive implementation was replaced with the slice's built-in `.len()` method. This improves efficiency from O(n) to O(1), prevents stack overflow for large inputs, and eliminates the unreachable panic case.
//	</comment>
//

#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum List {
    A,
    B,
    C,
    D,
}

fn length(a1: &[List]) -> i32 {
    a1.len() as i32
}